pkg/config/types/jobdefaults.go

```diff
@@ -47,7 +47,18 @@ type ResourcesConfig struct {
 }
 
 type DefaultPublisherConfig struct {
-	Config models.SpecConfig `yaml:"Config,omitempty"`
+	// Type specifies the publisher type. e.g. "s3", "local", "ipfs", etc.
+	Type string `json:"Type" yaml:"Type,omitempty"`
+
+	// Params specifies the publisher configuration data.
+	Params map[string]interface{} `json:"Params,omitempty" yaml:"Params,omitempty"`
+}
```

Comment:
You don't need to introduce this type. Just use `models.SpecConfig` directly

Reply:
We do in order to provide accurate comments on the type. The descriptions in `pkg/config/types/generated_descriptions.go` are generated from these code comments for convenience.